### ISSUE TITLE
Which method gives the highest accuracy on the IMDb dataset.

### ISSUE THREAD

**Issue by NagabhushanS** [keras-team/keras#8634]
The examples directory in keras repo has Python scripts giving various accuracies on the IMDb dataset. But which of them gives the highest accuracy on the IMDb dataset?. It would be helpful if the accuracies are commented on the top of the scripts for specific hyperparameters like epochs, optimizer etc.

**Comment by NagabhushanS** [comment]
Some of the files do contain the accuracies on the test  dataset. I noticed that the CNN based method is actually giving more accuracy than the bidirectional lstm, the lstm and the CNN+lstm based methods. Isn't this behaviour quite contradicting?

**Comment by FrancoSwiss** [comment]
Francois Chollet (creator of Keras) writes in his book on page 206: "LSTM's aren't helpful for sentiment-analysis problem ....there are more difficult NLP problems out there, where the strength of LSTM will become apparent."

Word order is of limited importance, so bi-directional LSTM's are not useful.

Attention: this is a small dataset. Neural networks are hungry and perform better with more data. Much more. Additionally, we have to be careful with comparisons: are we comparing apples with apples? Is there a cross validation? If yes, 5x? 10x? This can easily knock down your accuracy by >5%.

to my knowledge, accuracy:
- around 89% for Vanilla NN , CNN's and Logistic Regression (see below)

https://gist.github.com/prinsherbert/92313f15fc814d6eed1e36ab4df1f92d

**Comment by ShankulShukla** [comment]
https://github.com/ShankulShukla/Sentiment-analysis
